Testing/Functional/Products/TestPlans/teams.testplan.yaml

```diff
@@ -77,315 +65,367 @@ TestPlan:
+        Preconditions:
+          - Command: Set-CsTeamsMeetingPolicy
+            Splat:
+              Identity: Global
+              AutoAdmittedUsers: EveryoneInSameAndFederatedCompany
+        Postconditions: []
+        ExpectedResult: true
+      - TestDescription: MS.TEAMS.1.4v1 Compliant - AutoAdmittedUsers = EveryoneInCompanyExcludingGuests
+        Preconditions:
+          - Command: Set-CsTeamsMeetingPolicy
+            Splat:
+              Identity: Global
+              AutoAdmittedUsers: EveryoneInCompanyExcludingGuests
+        Postconditions: []
+        ExpectedResult: true
+  - PolicyId: MS.TEAMS.1.5v1
     TestDriver: RunScuba
     Tests:
-    - Postconditions: []
-      ExpectedResult: false
-      Preconditions:
-      - Command: Set-CsTenantFederationConfiguration
-        Splat:
-          AllowTeamsConsumer: true
-          AllowTeamsConsumerInbound: true
-      TestDescription: AllowTeamsConsumerInbound; AllowTeamsConsumer
-    - Postconditions: []
-      ExpectedResult: true
-      Preconditions:
-      - Command: Set-CsTenantFederationConfiguration
-        Splat:
-          AllowTeamsConsumer: false
-          AllowTeamsConsumerInbound: true
-      TestDescription: AllowTeamsConsumerInbound; Not AllowTeamsConsumer
-    - Postconditions: []
-      ExpectedResult: true
-      Preconditions:
-      - Command: Set-CsTenantFederationConfiguration
-        Splat:
-          AllowTeamsConsumer: true
-          AllowTeamsConsumerInbound: false
-      TestDescription: Not AllowTeamsConsumerInbound; AllowTeamsConsumer
-    - Postconditions: []
-      ExpectedResult: true
-      Preconditions:
-      - Command: Set-CsTenantFederationConfiguration
-        Splat:
-          AllowTeamsConsumer: false
-          AllowTeamsConsumerInbound: false
-      TestDescription: Not AllowTeamsConsumerInbound; Not AllowTeamsConsumer
-  - PolicyId: MS.TEAMS.5.2v1
+      - TestDescription: MS.TEAMS.1.5v1 Non-compliant - AutoAdmittedUsers = InvitedUsers
+        Preconditions:
+          - Command: 'Get-CsTeamsMeetingPolicy | % {Set-CsTeamsMeetingPolicy -Identity $($_.Identity) -AutoAdmittedUsers InvitedUsers }'
+          - Command: 'Get-CsTeamsMeetingPolicy | % {Set-CsTeamsMeetingPolicy -Identity $($_.Identity) -AllowPSTNUsersToBypassLobby $true }'
+        Postconditions: []
+        ExpectedResult: false   
+      - TestDescription: MS.TEAMS.1.5v1 Non-compliant - AutoAdmittedUsers = Everyone
+        Preconditions:
+          - Command: 'Get-CsTeamsMeetingPolicy | % {Set-CsTeamsMeetingPolicy -Identity $($_.Identity) -AutoAdmittedUsers Everyone }'
+          - Command: 'Get-CsTeamsMeetingPolicy | % {Set-CsTeamsMeetingPolicy -Identity $($_.Identity) -AllowPSTNUsersToBypassLobby $true }'
+        Postconditions: []
+        ExpectedResult: false        
+      - TestDescription: MS.TEAMS.1.5v1 Non-compliant - AutoAdmittedUsers = OrganizerOnly
```

Comment:
Will revisit policy 1.5 test after its planned change but note that there are a couple combinations of AutoAdmittedUsers and AllowPSTNUsersToBypassLobby are not allowed (overriden) and can be deleted from the combination of tests to be performed. For example Non-Complaint test where AutoAdmittedUsers is OrganizerOnly - will auto set AllowPSTNUsersToBypaasLobby to false (no option to make it true). Similarly the combo of 'Everyone' and 'false' is not allowed. Think those two combinations can be removed from the test cases.

Reply:
So still having some issues.  Looks like I was seeing other problems with the 1.5 tests using Nandaa updated rego code.  It looks like the complaint tests were supposed to have an expected return value of true.  But not 100 percent sure my understanding of the test framwork is correct.  I thought that compliant tests should have an expected return of true and non-compiant false respetively.  But this pattern is not followed in tests 4.1 and following.   For example the first 4.1 test is non-complaint and expects false but gets a return true from the rego code.

Reply:
Removed invalid test cases from 1.5.